lib/Service/SubmissionService.php

```diff
@@ -367,4 +377,39 @@ private function validateDateTime(string $dateStr, string $format) {
 		$d = DateTime::createFromFormat($format, $dateStr);
 		return $d && $d->format($format) === $dateStr;
 	}
+
+	/**
+	 * Validate short question answers if special validation types are set
+	 */
+	private function validateShortQuestion(array $question, string $data): bool {
+		if (!isset($question['extraSettings']) || !isset($question['extraSettings']['validationType'])) {
+			// No type defined, so fallback to 'text' => no special handling
+			return true;
+		}
+
+		switch ($question['extraSettings']['validationType']) {
+			case 'email':
+				return $this->mailer->validateMailAddress($data);
+			case 'number':
+				return is_numeric($data);
+			case 'phone':
+				// some special characters are used (depending on the locale)
+				$sanitized = str_replace([' ', '(', ')', '.', '/', '-', 'x'], '', $data);
+				// allow leading + for international numbers
+				if (str_starts_with($sanitized, '+')) {
+					$sanitized = substr($sanitized, 1);
+				}
+				return preg_match('/^[0-9]{3,}$/', $sanitized) === 1;
+			case 'regex':
+				// empty regex matches everything
+				if (!isset($question['extraSettings']['validationRegex'])) {
+					return true;
+				}
+				return preg_match($question['extraSettings']['validationRegex'], $data) === 1;
+			default:
+				$this->logger->error('Invalid input type for question detected, please report this issue!', ['validationType' => $question['extraSettings']->validationType]);
```

Comment:
```suggestion
				$this->logger->error('Invalid input type for question detected, please report this issue!', ['validationType' => $question['extraSettings']['validationType']]);
```
Another one that slipped through...